### DISCUSSION TITLE
Notifications to Discord

### DISCUSSION THREAD

**Discussion by ManiacMinor** [vogler/free-games-claimer#153] [Q&A] [answered]
I managed to get everything working but notifications. Looking for some help with this, thanks.

I don't understand how to set the notifications with Apprise.
My goal is to have this connected to a Discord webhook.

On Windows and cloned the repo.

I read the readme, and it set to set "NOTIFY to the notification services you want to use".
I tried changing lines in the config.js but clearly don't understand.

Below is what I tried:
```
notify: process.env.NOTIFY == 'discord://{WebhookID}/{WebhookToken}',
notify: 'discord://{WebhookID}/{WebhookToken}',
```
This is my first time using .js and apprise.

Thanks

**Comment by spider1163** [comment, marked as answer]
Hello,

Options are set via environment variables

Without Docker : 
https://github.com/vogler/free-games-claimer#without-docker

put options in data/config.env : 
NOTIFY='discord://{WebhookID}/{WebhookToken}/'
https://github.com/caronc/apprise/wiki/Notify_discord

**Comment by ManiacMinor** [comment]
This works, but the notify-test.js does not. I assume because I have no games to claim?

`error: Command failed: apprise discord://xxxx/xxxxx/ -i html '' -b 'epic-games:<br>- <a href="https://gaming.amazon.com/home">Epistory - Typing Chronicles</a> (claimed)'
The system cannot find the file specified.`

On line 11 in notify-test.js, if I remove everything after "epic-games:" the notification goes through as "'epic-games:'". 
I am unsure what file the error is having issues with.
`  notify('epic-games:<br>${html_game_list(notify_games)}');`

Thanks again!

**Reply by vogler** [reply]
You should just have to set the `NOTIFY` environment variable. No need to touch .js files.

`notify-test.js` will send a test notification (Epistory) to `NOTIFY`, content formatted via `html_game_list`, same way as the other scripts use it.

I don't know why the notification should work if you remove the HTML part of the body, the quoting looks fine.
I assume you have `apprise` installed. I don't know what `The system cannot find the file specified.` may refer to on Windows (command or just arg?).
Try running `apprise 'discord://xxxx/xxxxx/' -i html '' -b 'epic-games:<br>- <a href="https://gaming.amazon.com/home">Epistory - Typing Chronicles</a> (claimed)'` in the terminal and see what you have to remove to make it work.

**Comment by ManiacMinor** [comment]
It have messed around and if I use '' around 
`epic-games:<br>- <a href="https://gaming.amazon.com/home">Epistory - Typing Chronicles</a> (claimed)` 

I get output: 

> epic-games:
> - Epistory - Typing Chronicles (claimed)

This doesn't use the HTML formatting for a href. I tested it as a Windows notification as well, and does same
For my use I could change it since I don't think Discord can use HTML masking *yet* and works differently

Eg.
`apprise "discord://xxxxx/xxxxx/" -i html "" -b "Epic-Games: https://gaming.amazon.com/home<br>- Epistory - Typing Chronicles (claimed)"`

Output:

> epic-games: [https://gaming.amazon.com/home](https://gaming.amazon.com/home)
> - Epistory - Typing Chronicles (claimed)

**Reply by ManiacMinor** [reply]
Could you explain how to get it for work like I formatted, like what files to change and to what?
I am assuming line 239 in epic-games.js? But unsure how to change it exactly

Thanks for the help!

**Reply by vogler** [reply]
Interesting that discord doesn't support HTML.
You can try `markdown` as format to make links:
https://github.com/caronc/apprise/wiki/Notify_discord
https://www.reddit.com/r/discordapp/comments/fumcsq/how_do_i_make_text_a_clickable_linkhyperlink_on/

Just this line needs to be adjusted:
https://github.com/vogler/free-games-claimer/blob/7e4770b846135f7d0c36ac101914d35506f79474/util.js#L122

Could make a `game_list` that formats it as `html/markdown/text` depending on what the service supports. However, such a per-service conversion seems like an abstraction that should be added to apprise itself.
